store/src/java/com/zimbra/cs/mailbox/LocalMailboxLock.java

```diff
@@ -43,233 +35,47 @@
  *
  */
 public final class LocalMailboxLock implements MailboxLock {
```

Comment:
You deleted a ton of code from this class. I've not read through it all, but I'm worried that you changed the semantics of how this class works.

Reply:
There is a lot of logic in Zimbra's lock/release methods that looks important, like we don't want to change that. It's critical that this class behave EXACTLY as it did before since this is the default implementation of locking in the Zimbra product and needs to be 100% backwards compatible.

Reply:
redirected old lock implementation, please take a look